components/brave_wallet/browser/solana_transaction.h

```diff
@@ -44,8 +44,32 @@ class SolanaTransaction {
       mojom::SolanaTxDataPtr solana_tx_data);
   static absl::optional<SolanaTransaction> FromValue(const base::Value& value);
 
+  std::string to_wallet_address() const { return to_wallet_address_; }
+  std::string spl_token_mint_address() const { return spl_token_mint_address_; }
+  mojom::TransactionType tx_type() const { return tx_type_; }
+  uint64_t lamports() const { return lamports_; }
+  uint64_t amount() const { return amount_; }
+
+  void set_to_wallet_address(const std::string& to_wallet_address) {
+    to_wallet_address_ = to_wallet_address;
+  }
+  void set_spl_token_mint_address(const std::string& spl_token_mint_address) {
+    spl_token_mint_address_ = spl_token_mint_address;
+  }
+  void set_tx_type(mojom::TransactionType tx_type);
+  void set_lamports(uint64_t lamports) { lamports_ = lamports; }
+  void set_amount(uint64_t amount) { amount_ = amount; }
+
  private:
   SolanaMessage message_;
+
+  // Data fields to be used for UI, they are filled currently when we create
+  // SolanaTxData to transfer SOL or SPL tokens for UI.
+  std::string to_wallet_address_;
+  std::string spl_token_mint_address_;
+  mojom::TransactionType tx_type_ = mojom::TransactionType::Other;
+  uint64_t lamports_ = 0;  // amount of lamports to transfer
+  uint64_t amount_ = 0;    // amount of SPL tokens to transfer
```

Comment:
This will be used like this in the future:
Transferring SOL: https://github.com/brave/brave-core/commit/e4f972195a73dec0d58f86346db5b98c126ec8ab#diff-2042ad5dd3965e1a3c1a30ffe42f6610b0975da8a8a89f9dca00fd92549901f4R249-R251
Transferring SPL token: https://github.com/brave/brave-core/commit/e4f972195a73dec0d58f86346db5b98c126ec8ab#diff-2042ad5dd3965e1a3c1a30ffe42f6610b0975da8a8a89f9dca00fd92549901f4R346-R352